slick/src/main/scala/akka/stream/alpakka/slick/javadsl/package.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.slick.javadsl
+
+import slick.basic.DatabaseConfig
+import slick.jdbc.JdbcBackend
+import slick.jdbc.JdbcProfile
+import slick.jdbc.PositionedResult
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+
+/**
+ * Java API: Represents an "open" Slick database and its database (type) profile.
+ *
+ * <b>NOTE</b>: these databases need to be closed after creation to
+ * avoid leaking database resources like active connection pools, etc.
+ */
+sealed trait SlickSession {
+  val db: JdbcBackend#Database
+  val profile: JdbcProfile
+
+  /**
+   * You are responsible for closing the database after use!!
+   */
+  def close(): Unit = db.close()
+}
+
+/**
+ * Java API: Methods for "opening" Slick databases for use.
+ *
+ * <b>NOTE</b>: databases created through these methods will need to be
+ * closed after creation to avoid leaking database resources like active
+ * connection pools, etc.
+ */
+object SlickSessions {
```

Comment:
I wonder if changing the trait to an abstract class as Patrick suggests would allow this object to remain SlickSession, even for scala 2.11 from java? (probably not)

Reply:
object and abstract class is fine, trait + object is the problem in 2.11

Reply:
Ah, cool.

Reply:
Yeah, I fixed this using an abstract class so the Java DSL is once again using the same names as the Scala DSL. Thanks for the better solution!